pkg/sos/sos.go

```diff
@@ -0,0 +1,34 @@
+package sos
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	awscredentials "github.com/aws/aws-sdk-go-v2/credentials"
+)
+
+func NewSOSClient(ctx context.Context, zone, exoAPIKey, exoAPISecret string) (*s3.Client, error) {
+	sosEndpoint := "https://sos-" + zone + ".exo.io"
+	cfg, err := awsconfig.LoadDefaultConfig(
```

Comment:
Is the terraform provider only supposed to work with production endpoints? It would be nice if it did support preprod/dev environments too

Reply:
good point, I implemented a provider configuration attribute `sos_endpoint` now that allows to set the endpoint.